### ISSUE TITLE
sample_indices samples with replacement, biasing PPO minibatch gradients

### ISSUE THREAD

**Issue by CarsonBurke** [CarsonBurke/trading_bot_0#17]
## Bug Description

`sample_indices` in both `trading_bots/src/gym/base/memory.rs` (lines 9-20) and `trading_bots/src/burn/agent/base/memory.rs` independently samples `size` random indices using `rng.gen_range(0..indices.len())` in a loop with no deduplication. This means sampling is **with replacement**: the same experience can appear multiple times in a single minibatch while other experiences are never seen.

## Impact

In standard PPO, each epoch should shuffle the replay buffer and iterate over non-overlapping minibatches so that every experience is used exactly once per epoch. The current implementation introduces bias into minibatch gradient estimates because:

1. Some experiences are over-represented (sampled multiple times), giving them outsized influence on parameter updates
2. Other experiences are never sampled, wasting collected data and reducing sample efficiency
3. The variance of gradient estimates increases compared to the shuffle-and-partition approach

This affects the burn PPO agent directly at `trading_bots/src/burn/agent/ppo/agent.rs:91` where `sample_indices` is called in the training loop.

## Current Code

```rust
pub fn sample_indices(indices: MemoryIndices, size: usize) -> MemoryIndices {
    let mut rng = rand::thread_rng();
    let mut sample = Vec::<usize>::new();
    for _ in 0..size {
        unsafe {
            let index = rng.gen_range(0..indices.len());
            sample.push(*indices.get_unchecked(index));
        }
    }
    sample
}
```

## Suggested Fix

Replace with a Fisher-Yates shuffle and take the first `size` elements, or use `rand::seq::SliceRandom::choose_multiple` (without replacement). The caller in the PPO training loop should shuffle once per epoch and iterate non-overlapping chunks of `batch_size`.

## Files Affected

- `trading_bots/src/gym/base/memory.rs:9-20`
- `trading_bots/src/burn/agent/base/memory.rs:10-21` (duplicate implementation)
- `trading_bots/src/burn/agent/ppo/agent.rs:89-91` (call site in training loop)
